src/main/java/org/terasology/customOreGen/VeinsStructureDefinition.java

```diff
@@ -495,39 +491,31 @@ public boolean advance(float tolerance) {
 
 
     private final class SolidSphereStructure implements Structure {
-        protected final Transform mat;
-        protected final Transform invMat;
+        protected final Matrix4f mat;
+        protected final Matrix4f invMat;
         private Vector3i minPosition;
         private Vector3i maxPosition;
         private Vector3ic chunkSize;
         private Random random;
 
-        public SolidSphereStructure(Transform transform, Vector3ic chunkSize, Random random) {
+        public SolidSphereStructure(Matrix4f transform, Vector3ic chunkSize, Random random) {
             this.chunkSize = chunkSize;
             this.random = random;
             // build transformed bounding box from the local BB for a unit sphere
             float rMax = blockRadiusMultiplier.getMax();
             if (rMax < 0) {
                 rMax = 0;
             }
-            float[] bb = new float[]{-rMax, -rMax, -rMax, rMax, rMax, rMax};
-            transform.transformBB(bb);
-            float minX = Math.min(bb[0], bb[3]);
-            float minY = Math.min(bb[1], bb[4]);
-            float minZ = Math.min(bb[2], bb[5]);
-
-            minPosition = new Vector3i(Math.roundUsing(minX, RoundingMode.FLOOR), Math.roundUsing(minY, RoundingMode.FLOOR), Math.roundUsing(minZ, RoundingMode.FLOOR));
-
-            float maxX = Math.max(bb[0], bb[3]);
-            float maxY = Math.max(bb[1], bb[4]);
-            float maxZ = Math.max(bb[2], bb[5]);
 
-            maxPosition = new Vector3i(Math.roundUsing(maxX + 1, RoundingMode.FLOOR), Math.roundUsing(maxY + 1, RoundingMode.FLOOR), Math.roundUsing(maxZ + 1, RoundingMode.FLOOR));
+            AABBf aabb = new AABBf(-rMax, -rMax, -rMax, rMax, rMax, rMax);
+            aabb.transform(transform);
+            minPosition = new Vector3i(Math.roundUsing(aabb.minX, RoundingMode.FLOOR), Math.roundUsing(aabb.minY, RoundingMode.FLOOR), Math.roundUsing(aabb.minZ, RoundingMode.FLOOR));
```

Comment:
Please keep the line length in mind here :nerd_face: 

```suggestion
            minPosition = new Vector3i(
                              Math.roundUsing(aabb.minX, RoundingMode.FLOOR),
                              Math.roundUsing(aabb.minY, RoundingMode.FLOOR),
                              Math.roundUsing(aabb.minZ, RoundingMode.FLOOR));
```